src/common/system/proc_parser.cc

```diff
@@ -774,5 +774,77 @@ StatusOr<absl::flat_hash_set<std::string>> ProcParser::GetMapPaths(pid_t pid) co
   return map_paths;
 }
 
+StatusOr<absl::flat_hash_set<ProcParser::ProcessMap>> ProcParser::GetMapEntries(
+    pid_t pid, std::string libpath) const {
+  absl::flat_hash_set<ProcParser::ProcessMap> map_entries;
+
+  static constexpr int kProcMapNumFields = 6;
+
+  const std::filesystem::path proc_pid_maps_path = ProcPidPath(pid) / "maps";
+  PL_ASSIGN_OR_RETURN(std::string content, px::ReadFileToString(proc_pid_maps_path));
+  std::ifstream ifs;
+  ifs.open(proc_pid_maps_path);
```

Comment:
Looks like we're both using `ReadFileToString` and `ifs.open()`. Isn't that redundant?